final_task/rss_reader/rss_reader.py

```diff
@@ -0,0 +1,132 @@
+
+    elif (args.tohtml or args.topdf):
+
+        # set object of html converter library
+        doc = dominate.document(title="HTML document")
+        random_name = random.randint(1, 120)
+
+        for index, item in enumerate(channel.entries):
+            if (index == limit):
+                break
+            
+            media_content = parse_news.checkMediaContent(item)
+          
+            # create html structure
+            with doc:
+                with div():
+                    h2(item.title)
+                    p("Date: " + item.published)
+                    if (media_content):
+                        img(src=media_content)
+                    p("Description: " + parse_news.getDescription(item.description))
+
+        if (args.tohtml):
+           converter.intoHTML(args.tohtml, doc, random_name)
+
+        elif (args.topdf):
+            converter.intoPDF(args.topdf, doc, random_name)
+
+    else:
+
+        for index, item in enumerate(channel.entries):
+            if (index == limit):
+                break
+
+            # create folder for cache
+            today = date.today().strftime("%Y%m%d")
+            if not os.path.exists('cache/'):
+                os.makedirs('cache/')
+            cache_file = 'cache/' + today + '.txt'
+
+            # cache news
+            with open(cache_file, 'a', encoding="utf-8") as f:
+                f.write("Title: " + item.title + '\n' +
+                        "Link: " + item.link + '\n' +
+                        "Description: " + parse_news.getDescription(item.description) + '\n')
+     
+            # log news
+            logger.logging.debug("Title: " + str(item.title))
+            logger.logging.debug("Date: " + str(item.published))
+            logger.logging.debug("Link: " + str(item.link))
+            logger.logging.debug("Description: " + parse_news.getDescription(item.description))
+            logger.logging.debug("Links:"+"\n[1]: " + str(item.link) +
+                        "(link)\n[2]: " + str(parse_news.checkMediaContent(item)))
+
+            if (args.json):
+                print(parse_news.intoJson(item))
+
+            # print news
+            print("Title: ", item.title)
+            print("Date: ", item.published)
+            print("Link: ", item.link, '\n')
+            print("Description: ", parse_news.getDescription(item.description), '\n')
+            print("Links:", "\n[1]: ", item.link, "(link)\n[2]: ",
+                parse_news.checkMediaContent(item), '\n')
+
+except ValueError as e:
+    print("--limit or --date argument should be number")
+    logger.logging.error("ValueError: " + str(e))
+
+except AttributeError as e:
+    print("AttributeError: " + str(e))
+    print("Maybe check your source")
+    logger.logging.error("AttributeError: " + str(e))
+
+# fast paste in argument
+# source = "https://news.yahoo.com/rss"
+# source1 = "https://news.google.com/rss"
+# source2 = "https://www.theguardian.com/world/rss"
```

Comment:
It is better to split logic in this module into functions and use `if __name__ == "__main__"` here. Consider what will happen if I will try to import your module `rss_reader.py` to mine.

Reply:
Fixed